homeassistant/components/anova/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow for Anova."""
+from __future__ import annotations
+
+from anova_wifi import AnovaApi, AnovaOffline, InvalidLogin, NoDevicesFound
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN
+
+
+class AnovaConfligFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Sets up a config flow for Anova."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            api = AnovaApi(
+                aiohttp_client.async_get_clientsession(self.hass),
+                user_input[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+            )
+            try:
+                await api.authenticate()
+            except AnovaOffline:
+                errors["base"] = "cannot_connect"
+            except InvalidLogin:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            else:
+                try:
+                    devices = await api.get_devices()
```

Comment:
This is happening outside an exception catch and it looks like its doing asyncio so it might raise

Reply:
It is in its own try-except block, should it instead be in the first try-except block?

Reply:
I'd probably move `devices = await api.get_devices()` right under `await api.authenticate()` and collapse this up into the other try block since you have nice named exceptions and having two try blocks is a bit overkill